### DISCUSSION TITLE
Speaker control, possible or not?

### DISCUSSION THREAD

**Discussion by davidmacdonald80** [pyusb/pyusb#510] [Q&A] [answered]
I have an old usb speaker, Logitech Z305. It has 3 buttons on it, volume down, power or mute, and volume up.  I was hoping I could find a way to control/manage the power/mute button on it on the software side instead of the physical side. 
I was able to get the device assigned with pyusb, but I wasn't able to find any difference in the dev when the power/mute was on or off.  I used a basic loop to look at all the settings in the dev in each state. I also tried lsusb -v and still couldn't find any difference.  I also checked the cfg from both get_active_configuration() and get_configuration().

Edit:
I haven't had much time to test tonight.  I'm just wondering if I'm thinking about this wrong.  I know I can start the volume with the player at whatever I want.  I don't remember seeing volume levels in the cfgs, so could the power/mute button be like that?  once it's triggered it stays at the last known but doesn't set a cfg option because it's not really static?

If anyone has any ideas, 
Thanks in advance!
Otherwise, sorry for the post.

**Comment by mcuee** [comment]
Not so sure what you want to achieve is possible or not. Even if it is possiblem it is not to change the USB configuration, rather it is the command you send to the device and the response from the device.

You may want to read the basic of USB here.
https://www.usbmadesimple.co.uk/

You may have to carry out reverse engineering in order to know about the device since you do not seem to know the communication protoocl of the device.
https://slomkowski.eu/tutorials/eavesdropping-usb-and-writing-driver-in-python/

**Comment by mcuee** [comment, marked as answer]
Anyway, I do not think pyusb is the right one for you, There is an old question asked in the Arch Linux forum. You may want to try that way or better ask in the support channel of your Linux distro (if you are using Linux).
https://bbs.archlinux.org/viewtopic.php?id=113300

**Reply by davidmacdonald80** [reply]
Thanks for the information!  Ya i'm on raspbien in the context I would be working on this.  Sounds like i've got a project to work on for a while. Very much appreciated!
